ConfigurationSystem/Agent/RucioSynchronizer.py

```diff
@@ -0,0 +1,359 @@
+                    else:
+                        try:
+                            client.delete_rse_attribute(rse, 'PrimaryDataSE')
+                        except RSEAttributeNotFound:
+                            pass
+                        except Exception as err:
+                            self.log.error('Cannot remove RSE attribute PrimaryDataSE for %s : %s' % (rse, str(err)))
+
+
+            # Collect the user accounts from Dirac Configuration and create user accounts in Rucio
+            self.log.info("Synchronizing accounts")
+            listAccounts = [str(acc['account']) for acc in client.list_accounts()]
+            listScopes = [str(scope) for scope in client.list_scopes()]
+            accounts = getAllUsers()
+            DNMapping = {}
+            for account in accounts:
+                dn = getUserOption(account, 'DN')
+                email = getUserOption(account, 'Email')
+                DNMapping[dn] = email
+                if account not in listAccounts:
+                    self.log.info('Will create %s with associated DN %s' % (account, dn))
+                    try:
+                        client.add_account(account=account, type='USER', email=email)
+                        listAccounts.append(account)
+                    except Exception as err:
+                        self.log.error('Cannot create account %s : %s' % (account, str(err)))
+                    try:
+                        client.add_identity(account=account, identity=dn, authtype='X509', email=email, default=True)
+                    except Exception as err:
+                        self.log.error('Cannot create account %s : %s' % (account, str(err)))
+                    for rse in rses:
+                        client.set_local_account_limit(account, rse, 1000000000000000)
+                else:
+                    try:
+                        client.add_identity(account=account, identity=dn, authtype='X509', email=email, default=True)
+                    except Duplicate:
+                        pass
+                    except Exception as err:
+                        self.log.error('Cannot create identity %s for account %s : %s' % (dn, account, str(err)))
+                scope = 'user.' + account
+                if scope not in listScopes:
+                    try:
+                        self.log.info('Will create scope %s' % scope)
+                        client.add_scope(account, scope)
+                        self.log.info('Scope %s successfully added' % scope)
+                    except Exception as err:
+                        self.log.error('Cannot create scope %s : %s' % (scope, str(err)))
+
+            # Collect the group accounts from Dirac Configuration and create service accounts in Rucio
+            groups = getAllGroups()
+            for group in groups:
+                dns = getDNsInGroup(group)
```

Comment:
This variable looks like it can also be dropped in favour of using `for dn in getDNsInGroup(group)`

Reply:
Done